WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt

```diff
@@ -0,0 +1,123 @@
+import javax.inject.Inject
+
+@Suppress("LongParameterList")
+class MySiteSourceManager @Inject constructor(
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    private val currentAvatarSource: CurrentAvatarSource,
+    private val domainRegistrationSource: DomainRegistrationSource,
+    private val dynamicCardsSource: DynamicCardsSource,
+    private val quickStartCardSource: QuickStartCardSource,
+    private val scanAndBackupSource: ScanAndBackupSource,
+    private val selectedSiteSource: SelectedSiteSource,
+    postCardsSource: PostCardsSource,
+    siteIconProgressSource: SiteIconProgressSource,
+    private val mySiteDashboardPhase2FeatureConfig: MySiteDashboardPhase2FeatureConfig
+) {
+    private val mySiteSources: List<MySiteSource<*>> = listOf(
+            selectedSiteSource,
+            siteIconProgressSource,
+            quickStartCardSource,
+            currentAvatarSource,
+            domainRegistrationSource,
+            scanAndBackupSource,
+            dynamicCardsSource,
+            postCardsSource
+    )
+
+    fun build(siteLocalId: Int?, coroutineScope: CoroutineScope): List<LiveData<out PartialState>> {
+        return if (siteLocalId != null) {
+            mySiteSources.map { source -> source.build(coroutineScope, siteLocalId).distinctUntilChanged() }
+        } else {
+            mySiteSources.filterIsInstance(SiteIndependentSource::class.java)
+                    .map { source -> source.build(coroutineScope).distinctUntilChanged() }
+        }
+    }
+
+    fun isRefreshing(): Boolean {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            mySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach {
+                if (it.isRefreshing() == true) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
+
+    fun refresh() {
+        if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+            refreshAllSources()
+        } else {
+            refreshSelectSources()
+        }
+    }
+
+    fun onResume(isFirstResume: Boolean) {
+        when (isFirstResume) {
+            true -> refreshSelectSources()
+            false -> if (mySiteDashboardPhase2FeatureConfig.isEnabled()) {
+                refreshAllSources()
+            } else {
+                refreshSelectSources()
+            }
+        }
+    }
+
+    fun clear() {
+        domainRegistrationSource.clear()
+        scanAndBackupSource.clear()
+        selectedSiteSource.clear()
+    }
+
+    private fun refreshAllSources() {
+        mySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { it.refresh() }
+    }
+
+    private fun refreshSelectSources() {
+        selectedSiteSource.updateSiteSettingsIfNecessary()
+        currentAvatarSource.refresh()
+        quickStartCardSource.refresh()
+    }
+
+    /* QUICK START */
+
+    fun refreshQuickStart() {
```

Comment:
Minor (🔍): Consider adding a test for this public function as well. It is the only function not covered, else we are at 100% test coverage for the manager! 🎉

Reply:
🙇  thanks. Added in 0f53f51